app/controllers/addresses_controller.rb

```diff
@@ -12,12 +12,15 @@ def new
 
   def create  
     @address = @user.addresses.new(address_params)
-    if @address.save
-      flash[:notice] = "Address created successfully!"
-      redirect_to user_addresses_path
-    else
-      flash.now.alert = "Not Created"
-      render :new
+    respond_to do |format|
+      if @address.save
+        flash[:notice] = "Address created successfully!"
+        format.html { redirect_to user_addresses_path }
+      else
+        flash.now.alert = "Not Created"
+        format.html { render :new }
+        format.js
```

Comment:
What will happen if your request format is `js` and address is saved successfully?

Reply:
js format is only handled in else condition i.e. when address is not saved. If address is saved then it will render to addresses path.